src/boot/replaceRevive.js

```diff
@@ -46,24 +46,31 @@ const replacer = function replacer(key, value) {
     return origValue;
   }
 
-  if (origValue instanceof ZulipVersion) {
-    return { data: value.raw(), [SERIALIZED_TYPE_FIELD_NAME]: 'ZulipVersion' };
-  } else if (origValue instanceof URL) {
-    return { data: origValue.toString(), [SERIALIZED_TYPE_FIELD_NAME]: 'URL' };
-  } else if (origValue instanceof GravatarURL) {
-    return { data: GravatarURL.serialize(value), [SERIALIZED_TYPE_FIELD_NAME]: 'GravatarURL' };
-  } else if (origValue instanceof UploadedAvatarURL) {
-    return {
-      data: UploadedAvatarURL.serialize(value),
-      [SERIALIZED_TYPE_FIELD_NAME]: 'UploadedAvatarURL',
-    };
-  } else if (origValue instanceof FallbackAvatarURL) {
-    return {
-      data: FallbackAvatarURL.serialize(value),
-      [SERIALIZED_TYPE_FIELD_NAME]: 'FallbackAvatarURL',
-    };
-  } else if (origValue instanceof Immutable.Map) {
-    return { data: value, [SERIALIZED_TYPE_FIELD_NAME]: 'ImmutableMap' };
+  switch (Object.getPrototypeOf(origValue)) {
+    // Flow bug: https://github.com/facebook/flow/issues/6110
+    case (ZulipVersion.prototype: $FlowFixMe):
+      return { data: value.raw(), [SERIALIZED_TYPE_FIELD_NAME]: 'ZulipVersion' };
+    case (URL.prototype: $FlowFixMe):
+      return { data: origValue.toString(), [SERIALIZED_TYPE_FIELD_NAME]: 'URL' };
+    case (GravatarURL.prototype: $FlowFixMe):
+      return { data: GravatarURL.serialize(value), [SERIALIZED_TYPE_FIELD_NAME]: 'GravatarURL' };
+    case (UploadedAvatarURL.prototype: $FlowFixMe):
+      return {
+        data: UploadedAvatarURL.serialize(value),
+        [SERIALIZED_TYPE_FIELD_NAME]: 'UploadedAvatarURL',
+      };
+    case (FallbackAvatarURL.prototype: $FlowFixMe):
+      return {
+        data: FallbackAvatarURL.serialize(value),
+        [SERIALIZED_TYPE_FIELD_NAME]: 'FallbackAvatarURL',
+      };
+    case (Immutable.Map.prototype: $FlowFixMe):
```

Comment:
This bit isn't NFC because `instanceof Immutable.Map` would accept an `Immutable.OrderedMap` (the one subclass `Immutable` has for that class), and this won't.

... Which is a fix to a latent bug! We don't currently use `OrderedMap`, but if we did, the old code would cheerfully serialize it, and then deserialize it as a plain `Immutable.Map`.

Reply:
Ah, right, good catch—I'll make a note in the commit message and remove the [nfc] mark.